### ISSUE TITLE
When autowire with name more than 1 param, wrong autowire!

### ISSUE THREAD

**Issue by machinewu** [achingbrain/node-wantsit#2]
Here are my test code:
```
var Autowire = require('wantsit').Autowire,
	Container = require('wantsit').Container;

var Foo = function() {
	this._bar1 = Autowire({name: 'xxx'});
	this._bar2 = Autowire({name: 'yyy'});
};

Foo.prototype.do1 = function() {
	this._bar1.sayHello()
};

Foo.prototype.do2 = function() {
	this._bar2.sayHello()
};

///////////////////////////////////
var Bar1 = function() {
};

Bar1.prototype.sayHello = function() {
	console.log('1111!')
};

///////////////////////////////////
var Bar2 = function() {
};

Bar2.prototype.sayHello = function() {
	console.log('2222!')
};

///////////////////////////////////
var container = new Container();
container.register('xxx', new Bar1());
container.register('yyy', new Bar2());

var foo = new Foo();
container.autowire(foo);

foo.do1();
foo.do2();
```

Normally, it should return
```
1111!
2222!
```
But the return is
```
2222!
2222!
```

**Comment by machinewu** [comment]
If I new a Autowire, it will be fine.
```
var Foo = function() {
    this._bar1 = new Autowire({name: 'xxx'});
    this._bar2 = new Autowire({name: 'yyy'});
};
```

**Comment by achingbrain** [comment]
Well that's definitely a bug.

**Comment by achingbrain** [comment]
This should be fixed in version 2.1.3
